File: blazerod/render/src/main/java/top/fifthlight/blazerod/extension/RenderPipelineSnippetExt.java
package top.fifthlight.blazerod.extension;

import org.jetbrains.annotations.NotNull;
import top.fifthlight.blazerod.model.resource.VertexType;

import java.util.Optional;

public interface RenderPipelineSnippetExt {
    @NotNull
    Optional<VertexType> blazerod$getVertexType();
}

File: blazerod/render/src/main/kotlin/top/fifthlight/blazerod/extension/RenderPipelineSnippetExt.kt
package top.fifthlight.blazerod.extension

import com.mojang.blaze3d.pipeline.RenderPipeline
import top.fifthlight.blazerod.extension.RenderPipelineSnippetExt

@Suppress("CAST_NEVER_SUCCEEDS")
val RenderPipeline.Snippet.vertexType
    get() = (this as RenderPipelineSnippetExt).`blazerod$getVertexType`()
